### ISSUE TITLE
Extended extractor is removing blank lines

### ISSUE THREAD

**Issue by iHiD** [exercism/snippet-extractor#22]
I don't think the extended extractor should remove blank lines (except for ones at the start and end of the file). It makes the code quite a lot messier to read IMO.

For example, without the blank lines, this becomes quite hard to parse.
<img width="456" alt="Screenshot 2021-05-07 at 01 52 49" src="https://user-images.githubusercontent.com/286476/117382677-f87b5a00-aed6-11eb-82ac-d1a22f37e87a.png">

**Comment by joshiraez** [comment]
I considered this when doing the initial implementation, because the original extractor kept the empty lines,

I thought that, in a file with 10 lines of code max, minimum of 3 would be blank lines, so I chose to remove them altogether. Just like comments, they really ate a lot of the information from the snippet which was not useful. I thought the lines were just the same.

An example:

```
public class Solution {

  public static void main(String[] args) {
    // Something something
    int a = doStuff()
    System.out.println(a)
  }
  
  public int doStuff() {
    //Return a 2
    return 4;
  }
 } 
 ```
 
 For this snippet, the extractor would extract (if we kept blank lines). Please note that skipped stuff would become a blank line by the extractor algorithm, because it is agnostic about what it skips or strips.
 ```
 public class Solution {

  public static void main(String[] args) {

    int a = doStuff()
    System.out.println(a)
  }
  
  public int doStuff() {

 ```

Without blank lines

 ```
 public class Solution {
  public static void main(String[] args) {
    int a = doStuff()
    System.out.println(a)
  }
  public int doStuff() {
    return 4;
  }
 } 
 ```
 
 As you see, we now have important details of the implementation, that we would have lost before. We won like 40% of the text.
 
 So I'd go against removing all blank lines, just for the sake of having more identifiable snippets, without less clutter.
 
 In the other hand, there must be consideration that this is not trivial to implement. We will have to distinguish between stripped lines and skipped lines, or we will get a LOT of blank lines where there was a line comment and stuff like that. I know how I could implement it, rather, I'm not sure if its worth the extra complexity

**Comment by ErikSchierboom** [comment]
We do gain information, but it does come at the expense of readability IMHO. I personally would lean towards having the blank lines included. As someone mentioned, replacing multiple consecutive blank lines to just one makes sense though.

**Comment by iHiD** [comment]
I think the whitespace is important for making the snippets easily viewable. Also, a bit part of these snippets is to have nice visuals, and losing all the whitespace really detracts from that. In fact I'd argue that it actively makes me want to click on those solutions less because they look badly structured etc. People can always click on the thing to look at the full version.

So I think we need to find some way to do this.

**Comment by joshiraez** [comment]
I'll be working on this later this afternoon then :)
